aerofiles/seeyou/reader.py

```diff
@@ -193,3 +193,90 @@ def decode_description(self, description):
             return None
 
         return description
+
+    def decode_taskname(self, fields):
+        if not fields:
+            raise ParserError('Name field must not be empty')
+
+        return fields[0]
+
+    def decode_tasklist(self, fields, waypoints):
```

Comment:
what you want to test is the `read()` method, correct? I'd suggest that you create a small CUP file, save it into the `data` folder (which should actually be called `fixtures`), read that file with the new `read()` method in a new test and then assert that the waypoints in that file were read correctly